utils/cache.js

```diff
@@ -59,25 +69,64 @@ const cachePool = (opt = { maximumSize: CACHE_SIZE_MB }) => {
     }
   };
 
-  return { get, set, hits, cacheStore };
+  return { get, set, evict, hits, cacheStore };
+};
+
+const cachedKeysStore = () => {
+  const keyStore = new Map();
+
+  const getCachedKeys = (modelKey) => {
+    if (keyStore.has(modelKey)) {
+      return [...keyStore.get(modelKey).keys()];
```

Comment:
- Are we storing a complex data type in keyStore or is it just going to be array of key strings? 
Can you please tell the structure of the value we're storing in keyStore.

Reply:
Although there won't be any significant performance improvement, changed the return type to Set.